src/pages/coursesListPage/components/coursesList/CoursesList.tsx

```diff
@@ -0,0 +1,75 @@
+import Hls from 'hls.js'
+import { useEffect, useState } from 'react'
+import ReactPaginate from 'react-paginate'
+
+import './CoursesList.styled.scss'
+
+import { usePreviewCourses } from '../../../../api/courses/queries/usePreviewCourses'
+import { Spinner } from '../../../../ui-base/spinner/Spinner'
+import { CourseSection } from '../courseSection/CourseSection'
+
+export const CoursesList = () => {
+  const { data, isLoading } = usePreviewCourses()
+  const [startItemOffset, setStartItemOffset] = useState(0)
+  const sortedCourses = data?.courses.sort((a, b) => b.rating - a.rating)
+  const numberOfItems = 10
+  const endItemOffset = startItemOffset + numberOfItems
+  const paginatedCoursesList = sortedCourses?.slice(startItemOffset, endItemOffset)
+  const pageCount = Math.ceil(sortedCourses?.length! / numberOfItems)
+
+  const handlePageClick = (selectedItem: { selected: number }) => {
+    const newOffset = (selectedItem.selected * numberOfItems) % sortedCourses!.length
+    setStartItemOffset(newOffset)
+  }
+
+  useEffect(() => {
+    if (paginatedCoursesList) {
+      const videos = document.querySelectorAll('video')
```

Comment:
The same idea as in previous recommendation: **`useRef`** hook is a good choice for this option.